### ISSUE TITLE
replace project invitation w/ ability for anyone to add obs to projects

### ISSUE THREAD

**Issue by loarie** [inaturalist/inaturalist#237]
IMO, the process of adding observations to projects isn't working well. Mainly because

1. for the community the constant barrage of invitations is spammy and 
1. for project managers its a constant struggle to get people to add observations to a project just so they can be 'collected' but not necessarily because the projects want any special permissions (ie (a) access to private location data, (b) whatever the project terms specify, & (c) ability to post messages to the project community). 

Here are some radical changes that attempt to address this:

1. Make it so ProjectObservations (observations in the project) do not require ProjectUsers (users that have joined the project), e.g. separate joining a project from having project observations
1. Let any logged_in user add any observation (ie from any user) to any project they have joined
1. ProjectObservations without ProjectUsers would show up in a project, e.g. could be searched, would be added to the ProjectList, would be influenced by project curator IDs, but because they lack a ProjectUsers the project would not have permission (a) access to private location data, (b) whatever the project terms specify, & (c) ability to post messages to the project community.
1. On joining a project, present ProjectUser preferences for agreeing to (a) sharing geoprivacy info, (b) project terms & (c) receiving project posts that could apply as a blanket to all their data in the project
1. Add User preferences for opting out such that 'only I can add my observations to projects'
1. Get rid of project invitations
![screen shot 2014-02-13 at 10 29 13 pm](https://f.cloud.github.com/assets/598026/2168506/63d772c8-9541-11e3-8dab-05db86496666.png)
In the above menu
* the behavior of 'remove' is unchanged (ie observation owner and project curators can remove observations from a project), 
* 'fill in' triggers project field modal (add project specific fields), & 
* 'join' triggers join modal (create a ProjectUser)

The join modal includes preference that enable the ProjectUser to give blanket permissions for a,b,c above to the project which would apply to any observation added to the project:
![screen shot 2014-02-13 at 11 32 03 pm](https://f.cloud.github.com/assets/598026/2168782/22bbb656-954a-11e3-91bb-792280d8bc9e.png)
Note: These blanket  permissions mean if I join a project and give (a) geoprivacy permissions & (b) agree to the terms, that project could then arbitrarily add my observations and get these permissions. So its important that this is clear on join and that ProjectUsers are alerted not just when new curators are added to a project but also when 
1. project rules change (e.g. scope changes from seahorses to all taxa), & 
2. terms change. 

ProjectUsers could also not agree to these blanket permissions but rather enable (a)geoprivacy/(b)terms permissions for individual observations. This is safer, but would mean ProjectUsers have to sign off on every observation if they want to give these permissions. Enabling permissions for individual observations could happen like this:
![screen shot 2014-02-13 at 11 30 29 pm](https://f.cloud.github.com/assets/598026/2168771/ee586c24-9549-11e3-96dc-5340a6334950.png)

When migrating over to this new system, it would be important that existing projects keep current ProjectUsers and ProjectObservations, but also that projects can't get (a)geoprivacy & (b)terms permissions on observations they'd subsequently add unless ProjectUsers opt in. To achieve this, set all existing ProjectUser blanket permissions to true for (c)project posts, and false for (a)geoprivacy & (b)terms. However, all grandfathered ProjectObservations will have individual preferences set to true for (a)geoprivacy & (b)terms. ProjectUsers would have to opt-in to apply these preferences to new ProjectObservations.

Lastly, we'd have to add language making it clear to project teams that not all ProjectObservations in their project can be used according to the (b)terms. Tools to easily identify which can and which can't such as an additional column in the project CSV export would be useful.

I think these changes will give project managers the ability to 'collect' observations in their projects without burdening the community with tons of project invitations that need to be addressed. I like the tools projects provide for viewing leaderboards, visualizing searching obs, and visualizing/searching the checklists they compile from the obs. I also like that people want to 'collect' and visualize my obs in their own projects but I don't want to deal with the hassle of responding to invitations or adding them myself.

Likewise, if I wanted to give any special permissions (a,b,c above) to the projects, I could join the project and give blanket permissions once if I trust them, or give permissions to my project_observations individually if I don't. And if for some reason it bothers me to have my observations appear in projects (even though I've granted no special permissions) I could opt out entirely and the system would work as it does.

**Comment by kueda** [comment]
@loarie I updated the spec, can you check it out? I don't really remember what `usage_according_to_project_terms` was all about. Do we still need it? Maybe agreeing to terms could just be for people who create ProjectUsers?

**Comment by loarie** [comment]
We could handle terms (or more strictly obs/photo licensing) in 'trust' as I mentioned here  #480, of if that seems to niche we could just keep it in ProjectUsers

**Comment by loarie** [comment]
Part of this will involve some sort of tool/job to set a project to turn observations appropriate for a project in to project observations - we discussed that in order for this not to overwhelm the site, there should probably be constraints on this - e.g. must have place, must have taxon etc.

**Comment by kueda** [comment]
Question: currently when you leave a project, it removes all your observations from that project. Should that still happen? Maybe it should only happen if you only allow obs to be added to projects you've joined? Kind of an edge case, I guess.

**Comment by loarie** [comment]
I think it should remove trust but not the obs

**Comment by kueda** [comment]
If any project member can add any observation to the project, should any member also be allowed to remove any observation from the project?

**Comment by loarie** [comment]
I was thinking that anyone can add or remove their own observations but the project team only can add/remove anyone's obs. Were you thinking any one can add/remove anyone's obs? Hadn't considered that

**Comment by kueda** [comment]
Happy to go with your assumption. I don't feel that strongly about it.

**Comment by loarie** [comment]
I think the aggregator looks good. 

Before releasing this, I think we should simplify projects/show? I propose:
1) limiting the items in the sidebar to the right of the map to 'members' & the export tool
2) moving the most recent journal post to the sidebar
3) adding the BioBlitz header to all projects - but have it reflect project observations and their users
4) replace the observation feed below the map with the more graphical Bioblitz version
![screen shot 2015-04-09 at 9 31 48 am](https://cloud.githubusercontent.com/assets/598026/7071636/57d6c3b2-de9b-11e4-8d71-91cc1c51164c.png)
Should I make a separate issue for this?

Likewise, two things not necessary for release but that I forsee being useful down the road:
1) adding multiple taxa to must be in taxon rule - e.g. must be in Amphibia or Reptilia
2) adding a parameter for 'not in project' to the observations search so we could link to a place to find appropriate observations not in the project (e.g. Reptiles or Amphibia from Texas not in Herps of Texas)

**Comment by zooga1961** [comment]
Question:  Geolocation  & privacy.   I am not sure I understand the changes.    So here is the concern.    The Cedar Ridge Preserve has rare wild orchids that only grow in our location and one other near central texas...somewhere in Big Bend I think. (the orchid is rare in Texas only as it is not on any list)   

The management at CRP  & the researchers don't want to reveal their exact location because they are worried about collectors. ( If you know what you are looking for and when to look... of course you can find them. )    

Currently we obscure the location for this orchid so we can include them in the CRP database.  and when I see one that is not obscured I contact the user and explain and request they obscure the location.   Most think it is crazy but comply... and it is what management at CRP & the folks researching these orchids want.. so we do it..    

With the changes; anyone who has joined the project can see the location even if obscured?  
If that is the case then CRP and the Researchers will NOT want us to include the observation for these orchids.   and they will want us to remove the existing observations....  yep .. that is how crazy they are here in Texas...

Deb

**Comment by naturalistcharlie** [comment]
My understanding is that people would NOT be able to access the obscured
location by adding the observation to their project, unless the user
specifically granted them that permission.

In addition, if the orchid species is one that is at high risk for
collection we can have it set to automatically obscure the location when
one is entered.

On Thu, Apr 9, 2015 at 1:43 PM, Deborah Nelson <notifications@github.com>
wrote:

> Question: Geolocation & privacy. I am not sure I understand the changes.
> So here is the concern. The Cedar Ridge Preserve has rare wild orchids that
> only grow in our location and one other near central texas...somewhere in
> Big Bend I think. (the orchid is rare in Texas only as it is not on any
> list)
>
> The management at CRP & the researchers don't want to reveal their exact
> location because they are worried about collectors. ( If you know what you
> are looking for and when to look... of course you can find them. )
>
> Currently we obscure the location for this orchid so we can include them
> in the CRP database. and when I see one that is not obscured I contact the
> user and explain and request they obscure the location. Most think it is
> crazy but comply... and it is what management at CRP & the folks
> researching these orchids want.. so we do it..
>
> With the changes; anyone who has joined the project can see the location
> even if obscured?
>
> If that is the case then CRP and the Researchers will NOT want us to
> include the observation for these orchids. and they will want us to remove
> the existing observations.... yep .. that is how crazy they are here in
> Texas...
>
> Deb
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/inaturalist/inaturalist/issues/237#issuecomment-91305874>
> .
>



-- 
============================
Charlie Hohn
Montpelier, Vermont
coyot.es/slowwatermovement

**Comment by loarie** [comment]
'With the changes; anyone who has joined the project can see the location even if obscured?' No, as before if you Join a project you are trusting them to see your observations private location data.  The only difference is project admins can add anyones observations. So if you don't join, they might add your observations but won't have access to your private location data. And if you do join, be advised they could potentially add and thus get access to private location data for any of your other observations - so best practice would be to leave projects you don't trust to be respectful of your private location data.

**Comment by naturalistcharlie** [comment]
So there will be no way to join a project without sharing data? I think it
might be better if I could join a project without sharing location data...
in many cases people set up projects where I would like to support their
interest in nature and iNaturalist, but I don't know a thing about them and
don't want to give them obscured data (or don't have permission to do so,
etc).  For that reason I think it should be possible to join projects
without sharing that unobscured data. It's not essential, just a thought.

On Thu, Apr 9, 2015 at 1:52 PM, Scott Loarie <notifications@github.com>
wrote:

> 'With the changes; anyone who has joined the project can see the location
> even if obscured?' No, as before if you Join a project you are trusting
> them to see your observations private location data. The only difference is
> project admins can add anyones observations. So if you don't join, they
> might add your observations but won't have access to your private location
> data. And if you do join, be advised they could potentially add and thus
> get access to private location data for any of your other observations - so
> best practice would be to leave projects you don't trust to be respectful
> of your private location data.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/inaturalist/inaturalist/issues/237#issuecomment-91309054>
> .
>



-- 
============================
Charlie Hohn
Montpelier, Vermont
coyot.es/slowwatermovement

**Comment by naturalistcharlie** [comment]
one other thing.. at least now you can choose to add only your unobscured
locations. Does this mean that if I have done that to a project they can
then get my obscured data from observations I didn't share with them? This
could be a big problem for me, I might have to quit a lot of projects if
that is the case. I'd urge you to do things differently if you can.

On Thu, Apr 9, 2015 at 1:55 PM, Charlie Hohn <naturalist.charlie@gmail.com>
wrote:

> So there will be no way to join a project without sharing data? I think it
> might be better if I could join a project without sharing location data...
> in many cases people set up projects where I would like to support their
> interest in nature and iNaturalist, but I don't know a thing about them and
> don't want to give them obscured data (or don't have permission to do so,
> etc).  For that reason I think it should be possible to join projects
> without sharing that unobscured data. It's not essential, just a thought.
>
> On Thu, Apr 9, 2015 at 1:52 PM, Scott Loarie <notifications@github.com>
> wrote:
>
>> 'With the changes; anyone who has joined the project can see the location
>> even if obscured?' No, as before if you Join a project you are trusting
>> them to see your observations private location data. The only difference is
>> project admins can add anyones observations. So if you don't join, they
>> might add your observations but won't have access to your private location
>> data. And if you do join, be advised they could potentially add and thus
>> get access to private location data for any of your other observations - so
>> best practice would be to leave projects you don't trust to be respectful
>> of your private location data.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/inaturalist/inaturalist/issues/237#issuecomment-91309054>
>> .
>>
>
>
>
> --
> ============================
> Charlie Hohn
> Montpelier, Vermont
> coyot.es/slowwatermovement
>



-- 
============================
Charlie Hohn
Montpelier, Vermont
coyot.es/slowwatermovement

**Comment by zooga1961** [comment]
I can't monitor everyone's ..projects and with whom they share their observations .. this opens up the orchid's location to other projects and other people when submitted by someone who doesn't know better.    

and again the researchers and management at CRP will want me to remove any observations  with this orchid.

**Comment by loarie** [comment]
charlie, with these changes you and the project team can add your observations to a project without joining it. So you could show support by adding your observations but not joining if you didn't want to share location data. 

You would only join a project if you wanted to commit to:
a) receiving project posts in your updates feed
b) granting them trust to access private location data of anything added to their project (by you or them)
c) having them show up in the 'your projects' section of the website and app

If you don't want to commit to a, b & c, leave or don't join the project. So yes, once these changes are in place you should leave any project you don't trust with your private location data. 

I'm in the same boat as you. I've joined over 200 projects not because of (a) wanting to receive their posts, (b) wanting to share my private location data with them or (c) wanting them in my 'your projects' tab but rather (d) because they asked me to add my observation and I didn't want to be rude. With these changes I will plan on leaving all but a few of these projects and they will still be able to include my observations (d) but I can excuse myself from (a), (b), and (c)

**Comment by loarie** [comment]
zooga1961 not sure I understand your concern, but let me know if my response to charlie above doesn't clarify. If you don't trust projects with your orchid data, you should probably leave them when these features are deployed

**Comment by naturalistcharlie** [comment]
Hmm, yeah I don't really see how this can work. It's not about 'trust' per
se but permission and caution. For instance, I added like 5000 observations
to that Great Nature project. I very carefully went through and excluded
the obscured ones.  It's not that I distrust or dislike national geographic
but I don't think they need precise locations of rare species, as they
don't have a genuine management need for that. Likewise there are some
observations that may be on private land I didn't want public. My
understanding was that as long as I didn't add them they would not have
access to them. This feels like a small violation of trust to me. I am now
going to have to pull 5000+ observations out of that project so I can quit
it, which is a bummer. I hope this is made clear to everyone before the
change. It seems like it will be necessary to have every person reaffirm
that they want to be a part of a project since they now have access to
other unobscured data the users did not add! This will create a big problem
when users who are no longer active will either no longer be in a project
at all, or else will without their knowledge have given their personal
information to someone they did not agree to do that with.

I know I get riled up about geoprivacy, and I'm sorry to come out about
this late in the planning process, but I think this is a pretty serious
concern here. If joining a project lets the project owner have ANY location
data, I really don't feel comfortable joining ANY projects except those run
by Heritage networks and such. For instance what's to stop the 'California
Pine Tree Project' from droppign their requirements and adding a Vermont
orchid? I don't distrust any one person, but I can't put my faith in a
system that has that huge of a possible security flaw. Not everyone is as
obsessive as me... most people would happily join a project to say, share
their California poppy sightings or something and this opens up the site to
possibility of very, very serious abuse.

In the LEAST you need to make sure the project has to re-request permission
if they expand their range of possible observations they can add.  In fact
I'd go so far as to say if this system is implemented I would not want
auto-obscured rare species data available to projects at all, unless they
are somehow certified as having a serious need for that data. For instance
there is no reason a Vermont project should be able to remove their
boundary and add my California observations of rare plants...

I hope others weigh in on this (without reigniting the 'just obscure
everything' thing). I see this as a really serious flaw in project security
that would cause me to stop participating in almost all of them.

Really concerned here. As you know I am a big open data advocate but TRUST
is very important with open data. I hope you take this concern seriously,
and thank you much for listening. Sorry for another long rant.

C



On Thu, Apr 9, 2015 at 2:07 PM, Scott Loarie <notifications@github.com>
wrote:

> charlie, with these changes you and the project team can add your
> observations to a project without joining it. So you could show support by
> adding your observations but not joining if you didn't want to share
> location data.
>
> You would only join a project if you wanted to commit to:
> a) receiving project posts in your updates feed
> b) granting them trust to access private location data of anything added
> to their project (by you or them)
> c) having them show up in the 'your projects' section of the website and
> app
>
> If you don't want to commit to a, b & c, leave or don't join the project.
> So yes, once these changes are in place you should leave any project you
> don't trust with your private location data.
>
> I'm in the same boat as you. I've joined over 200 projects not because of
> (a) wanting to receive their posts, (b) wanting to share my private
> location data with them or (c) wanting them in my 'your projects' tab but
> rather (d) because they asked me to add my observation and I didn't want to
> be rude. With these changes I will plan on leaving all but a few of these
> projects and they will still be able to include my observations (d) but I
> can excuse myself from (a), (b), and (c)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/inaturalist/inaturalist/issues/237#issuecomment-91313433>
> .
>



-- 
============================
Charlie Hohn
Montpelier, Vermont
coyot.es/slowwatermovement

**Comment by zooga1961** [comment]
I am not sure it does..     The issue is an orchid that only grows in two places in Texas, one being Cedar Ridge Preserve and the other being in Central Texas.     Management at Cedar Ridge Preserve and Researchers of the orchid would like it's location obscured  for fear of collection.     (and I had not really thought about this before now).... but other people observing this orchid and reporting it to other projects put it in jeopardy .     so can we do as Charlie first suggested   ... we can have it set to automatically obscure the location when any observation for this orchid is entered into iNat for the geolocation of Cedar Ridge Preserve?     I have asked Jim Varnum for a list. because I just reported an observation for a cousin of this orchid.. and that orchid will have the same issue.   so I know there are at least two species for which CRP and the researchers want private to protect it

**Comment by kueda** [comment]
Guys, this is getting beyond the technical requirements of this issue and requires a larger audience. Please continue this conversation in the Google Group, where there's already a thread for this: https://groups.google.com/d/topic/inaturalist/JGEpm2jSd3Y/discussion

**Comment by zooga1961** [comment]
my apologies.   and Thank you

**Comment by kueda** [comment]
Revised spec as per [discussion in the Google Group](https://groups.google.com/d/topic/inaturalist/JGEpm2jSd3Y/discussion) / re-interpretation of Scott's original intent (mea culpa):

- [ ] rename `usage_according_to_project_terms` to `coordinate_access` in ProjectObservation and ProjectUser
- [ ] for ProjectUser, `coordinate_access` defaults to true
- [ ] for ProjectObservation, `coordinate_access` defaults to the equivalent value for ProjectUser. If there is no ProjectUser, it defaults to false
- [ ] grandfathering in the migration: ensure project curators have access to coordinates of all existing project observations like it has been working, but not to any future contributions unless the observers enable it
  - [ ] set `ProjectUser#preferred_coordinate_access` to false for all existing project users
  - [ ] set `ProjectUser#preferred_coordinate_access` to true for all existing project observations
- [ ] base access to private coordinates of project observations on `prefers_coordinate_access?`
- [ ] ensure `project_post_updates` is working
- [ ] get a test machine up for people to check this out

**Comment by loarie** [comment]
Looking at test up on http://gorilla.inaturalist.org/, I think there's a bug where joined projects are not displaying on 'Add to project' menu in obs/show (they do appear on the 'Add to project' on obs/by_you#batch_edit). To recreate:
1) join a project
2) go to one of my observations and expand 'Add to project'
3) can't find project
![screen shot 2015-04-15 at 4 20 18 pm](https://cloud.githubusercontent.com/assets/598026/7171489/f687fd22-e38d-11e4-890c-701d1778979a.png)

**Comment by loarie** [comment]
Currently on obs/new, if there is a project param (https://www.inaturalist.org/observations/new?project_id=4026) but the user has not joined, we show this yellow checkbox to make sure someone joins before adding an obs. With these changes, since its no longer necessary to join to add an obs, I suggest removing the checkbox from obs/new
![screen shot 2015-04-15 at 2 53 08 pm](https://cloud.githubusercontent.com/assets/598026/7171825/8673629c-e392-11e4-9c81-4c5b86c8d001.png)

**Comment by loarie** [comment]
I suggest changing the language of the top setting to something like the following to make it clear that if I check this box project curators could see private coordinates of things they (ie not me) add to the project. This was not clear to me from the text as its written and seems to be the main voiced concern
![screen shot 2015-04-15 at 12 36 08 pm](https://cloud.githubusercontent.com/assets/598026/7171678/7e719eb2-e390-11e4-803e-959b062476c6.png)
1) **Allow project curators to view your private coordinates of anything added to the project**
*Remember, project curators can add any of your observations to their project unless you've opted out in your [account settings](http://www.inaturalist.org/people/477/edit). Only check this box if you trust the project to view your private coordinates of whatever might be added. Otherwise, you can still grant access to private coordinates on a observation-by-observation basis.*

**Comment by loarie** [comment]
I suggest changing this language to the following to emphasize that opting-out of this is not necessary to avoid opting-in (ie joining a project) to sharing your private coordinates/receiving updates. Otherwise I anticipate a number of people unnecessarily opting out of this do to confusion.
![screen shot 2015-04-15 at 4 27 12 pm](https://cloud.githubusercontent.com/assets/598026/7171760/b41a5cec-e391-11e4-9068-3228dc0a6150.png)
**Which Projects Can Add Your Observations?**
*Remember this does not give projects permission to access your private coordinates or send you updates. You must join projects in order to grant these permissions*

**Comment by loarie** [comment]
The language on leaving a project doesn't include what I think would be the most obvious preference which would be: 'revoke all access to private coordinates but leave the observations in the project'. To me the top choice seems kind of silly because why leave a project if you want them to still have access to your private coordinates, and the bottom choice is kind of silly because the project can just add them right back unless you opt out in account settings 
![screen shot 2015-04-15 at 12 33 41 pm](https://cloud.githubusercontent.com/assets/598026/7171518/4f6f92c4-e38e-11e4-8df0-dbc429b2ec04.png)
I'd suggest either removing the choice and just having it default to 'revoke all access to private coordinates but leave the observations in the project', but if others think these other 2 settings would be useful, how about all three, could be worded like (the top and bottom choices are unchanged just reworded):

**Leaving this project means the project curators will no longer be able to send you updates or see the private coordinates of your observations later added to the project. Choose to...**
1) have your observations stay in the project and allow the project curators to continue to see the private coordinates for these observations already added
2) have your observations stay in the project but revoke permissions for project curators to see your private coordinates of your observations already added to the project
3) remove all your observations from the project (remember you must change your [account settings](https://www.inaturalist.org/people/477/edit) if you want to opt out of letting project curators add your observations to projects)

**Comment by loarie** [comment]
Since there's no way for project curators to bulk remove obs form projects, I think we should either make the Aggregator (1) available only for trusted projects or (2) add an 'are you sure' alert to checking the aggregator.
![screen shot 2015-04-15 at 6 08 31 pm](https://cloud.githubusercontent.com/assets/598026/7172319/7bcf7be8-e39a-11e4-972e-44c3a085140c.png)
Maybe something like:
"This will automatically check for and add all observations of Reptiles or Amphibians from Texas to this project. Are you sure you want to do this?"

**Comment by loarie** [comment]
I don't think the Aggregator is working on http://gorilla.inaturalist.org/ - I tried enabling for a few projects and waited ~1hr and nothing changed.

**Comment by loarie** [comment]
Re: UI redesign
1) why do we need top contributors twice? Maybe get rid of this?
2) why do we need 'add from your observations?' after removing the inviter - this doesn't go anywhere useful
3) why do we need 'invite observations' after removing the inviter - this doesn't go anywhere useful
4) does anyone use 'user stats'? It might be kind of confusing since the #obs no longer reflects anything meaningful about 'success' of the project. 
5) Frustrating having the journal in two places, might be cleaner when no journal posts to have the button show up below the map.
Thoughts?
![screen shot 2015-04-17 at 3 56 35 pm](https://cloud.githubusercontent.com/assets/598026/7212360/6220c220-e51a-11e4-9241-2c54e879a3c3.png)

**Comment by kueda** [comment]
1. Fine.
2. I'll replace it with "Download CSV template" or something. That's the only reason I left that there.
3. As you know I hate that thing and am fine removing it, but people will flip again. I can't think of an efficient way to show obs that are *not* in a project.
4. I've certainly had complaints from people when it has stopped working in the past, so I left it. I don't think it's doing much harm.
5. Fine.

**Comment by loarie** [comment]
3 Since the inviter is gone, its confusing to leave a link with that name in that goes to obs search. Folks have already requested a way to show obs that are not in a project, but I think we can kick that down the road.
4. ok

**Comment by loarie** [comment]
Actually I see its only projects of type 'contest' and 'observation contest' that render the now redundant leaderboards. The 'normal' members UI is actually kind of nice. Maybe we just remove those two project types instead and set them to normal?
![screen shot 2015-04-17 at 4 21 10 pm](https://cloud.githubusercontent.com/assets/598026/7212564/1a2ed7b4-e51e-11e4-8582-af6fa1de862c.png)

**Comment by pleary** [comment]
I think we can use Elasticsearch now for finding observations in / not in a project. For example for the Barnstable project http://www.inaturalist.org/projects/biodiversity-of-barnstable-county-ma-usa :

```Observation.elastic_search(filters: [ { place: Place.find(1458) } ]).total_entries```

```Observation.elastic_search(filters: [
  { place: Place.find(1458) },
  { term: { project_ids: 487 } }
]).total_entries```

**Comment by loarie** [comment]
If I've joined a project and checked 'Allow project curators to view my private coordinates' its confusing that this box on the observations/show project observation settings modal is unchecked. Can it not appear in this case or appear checked?
![screen shot 2015-04-17 at 6 33 32 pm](https://cloud.githubusercontent.com/assets/598026/7214017/c8bf3320-e549-11e4-8bfa-c06b0236c9c9.png)

**Comment by loarie** [comment]
I think adding the more detailed language about privacy under 'Allow project curators access...' is good and relevant there since it presented as an option. But now the language below project curators 'Please be aware...unless you trust the project curators' is redundant. Maybe remove it?
![screen shot 2015-04-17 at 6 30 56 pm](https://cloud.githubusercontent.com/assets/598026/7214027/2f96c6d0-e54a-11e4-8b37-3374e6a29ece.png)

**Comment by loarie** [comment]
Since sharing private location data is presented as an option when joining a project I think the 'Joining the project will ... project terms.' language on the observations/show project observation settings modal should be removed
![screen shot 2015-04-17 at 6 29 59 pm](https://cloud.githubusercontent.com/assets/598026/7214037/aa0cace0-e54a-11e4-88e7-aeea54943a89.png)

**Comment by loarie** [comment]
as mentioned above, I suggest removing project type 'Contest' and 'Observation Contest' and replacing the text with 'Bioblitz projects occur over specified time periods and display observations slightly differently. Assessment projects display and link to associated species assessments that can be edited by the project team'
![screen shot 2015-04-17 at 5 16 14 pm](https://cloud.githubusercontent.com/assets/598026/7214041/221d377c-e54b-11e4-8942-236ea393dc46.png)

**Comment by loarie** [comment]
Since the project checklists no longer display, I suggest moving these sections to the 'admin only' section on projects/edit where they can be accessed in the off chance that someone still requires them.
![screen shot 2015-04-17 at 9 52 31 pm](https://cloud.githubusercontent.com/assets/598026/7214053/1b50ede8-e54c-11e4-8dbe-853a40734e95.png)
![screen shot 2015-04-17 at 9 51 05 pm](https://cloud.githubusercontent.com/assets/598026/7214054/1c81e0d2-e54c-11e4-94e3-437e8e80c0f9.png)

**Comment by loarie** [comment]
Suggest rewording 'so participants do not need...' to 'so observations don't need to be added to the project to drive the project stats'
![screen shot 2015-04-17 at 9 55 10 pm](https://cloud.githubusercontent.com/assets/598026/7214063/c5a71bf0-e54c-11e4-8058-ace5c7df15d6.png)

**Comment by loarie** [comment]
Suggest updating this modal text on projects/show for BioBlitz projects: 'This is to remove the hurdle of...' to 'This is to remove the hurdle of adding observations to a project to drive the stats in real time. Note that observations won't be available for export or display on various project views such as member contribution views on the website or project observation searches in the mobile apps unless the project team and participants actively add relevant observations like they would in a normal project.'
![screen shot 2015-04-17 at 9 55 28 pm](https://cloud.githubusercontent.com/assets/598026/7214079/a5dd4a3c-e54d-11e4-8dc0-65c5d4e95433.png)

**Comment by kueda** [comment]
Additional proposal by @loarie to satisfy project manager concerned about people opting out of sharing coordinates by default for fear of project managers adding observations they didn't want them to add to the project (and the assumption that they will not deal with this by granting curator coordinate access for individual project observations):

- [ ] Change the ProjectUser#curator_coordinate_access preference to include the options *observer* (label: "Allow project curators to view private coordinates of observations I add to this project", *any* (label: "Allow project curators to view your private coordinates of observations anyone adds to this project", and *none* (label:  "Don't share private coordinates of my observations by default")
- [ ] alter the logic that sets the default value for ProjectObservation#curator_coordinate_access to make a decision based on who is adding it to the project

Scott, correct me if I've gotten this wrong.

**Comment by loarie** [comment]
yes - and also:
Make the default checked option on join (and likewise the currently optionless-UI default on joining from the mobile apps)  be 'observer'
